templates/admin/mail-template/detail.html.twig

```diff
@@ -6,9 +6,30 @@
     {{ parent() }}
 
     <div class="mail-template--preview">
-        <a class="btn btn-secondary" href="{{ path('admin_mail_template_template_file', {id: entity.primaryKeyValue}) }}">{{ 'Show template file'|trans }}</a>
-        <a class="btn btn-primary" href="{{ path('admin_mail_template_preview', {id: entity.primaryKeyValue}) }}">{{ 'Preview template'|trans }}</a>
-        <a class="btn btn-primary" href="{{ path('admin_mail_template_data', {id: entity.primaryKeyValue}) }}">{{ 'Preview template data'|trans }}</a>
-        <a class="btn btn-primary" href="{{ path('admin_mail_template_preview', {id: entity.primaryKeyValue, with_data: true}) }}">{{ 'Preview template with data'|trans }}</a>
+        <form method="get">
+            <a class="btn btn-secondary" href="{{ path('admin_mail_template_template_file', {id: entity.primaryKeyValue}) }}">{{ 'Show template file'|trans }}</a>
+            <a class="btn btn-primary" href="{{ path('admin_mail_template_preview', {id: entity.primaryKeyValue}) }}">{{ 'Preview template'|trans }}</a>
+
+            {% if preview_entities %}
+                <button class="btn btn-primary" type="submit" formaction="{{ path('admin_mail_template_data', {id: entity.primaryKeyValue}) }}">{{ 'Preview template data'|trans }}</button>
+                <button class="btn btn-primary" type="submit" name="with_data" value="true" formaction="{{ path('admin_mail_template_preview', {id: entity.primaryKeyValue}) }}">{{ 'Preview template with data'|trans }}</button>
+
+                <div class="form-group">
+                    <label for="previewEntity">{{ 'Preview entity'|trans }}</label>
+                    <select id="previewEntity" class="form-control" name="entity[id]">
+                        {% for entity in preview_entities %}
```

Comment:
There is already an `entity` variable in use in here, consider choosing a different name for this, e.g. `preview_entity`.